### PR TITLE
Add documentation and docstrings, improve DIA-NN parsing, refactoring

### PR DESCRIPTION
**Pull request by Cajac102**

### CODE REVIEW

File: proteobench/modules/dda_quant_ion/dda_quant_ion_module.py [hunk 1]
```diff
@@ -66,31 +106,34 @@ def benchmarking(
...
             raise QuantificationError(f"Error generating quantification scores: {e}")
 
+        # generate intermediate data structure
         try:
             intermediate_data_structure = quant_score.generate_intermediate(standard_format, replicate_to_raw)
         except Exception as e:
             raise IntermediateFormatGenerationError(f"Error generating intermediate data structure: {e}")
 
-        # try:
+        #try:
```
Comment (RobbinBouwmeester) [inline proteobench/modules/dda_quant_ion/dda_quant_ion_module.py:123]: Uncomment

File: proteobench/modules/dda_quant_ion/dda_quant_ion_module.py [hunk 2]
```diff
@@ -66,31 +106,34 @@ def benchmarking(
...
         current_datapoint = Datapoint.generate_datapoint(
             intermediate_data_structure, input_format, user_input, default_cutoff_min_prec=default_cutoff_min_prec
         )
-        # except Exception as e:
+        #except Exception as e:
```
Comment (RobbinBouwmeester) [inline proteobench/modules/dda_quant_ion/dda_quant_ion_module.py:127]: Uncomment

File: proteobench/modules/quant_base/quant_base_module.py [hunk 3]
```diff
@@ -54,21 +61,24 @@ def is_implemented(self) -> bool:
         """Returns whether the module is fully implemented."""
         return True
 
-    EXTRACT_PARAMS_DICT = {
```
Comment (RobbinBouwmeester) [inline proteobench/modules/quant_base/quant_base_module.py:57]: Is this not needed anymore? Is this somewhere else?

**Comment by rodvrees** [comment]
@SamvPy @Cajac102 @Alirezak2n I added our names to the contribution, can you check if I added your stuff correctly?

**Review by RobbinBouwmeester — APPROVED** [review]

**Pull request merged**